osu.Game.Rulesets.Osu/Mods/OsuModRelax.cs

```diff
@@ -125,6 +131,20 @@ void changeState(bool down)
                 isDownState = down;
                 lastStateChangeTime = time;
 
+                // legacy replays do not contain key-presses with Relax mod, so they need to be triggered by themselves.
+                if (legacyReplay)
+                {
+                    if (!down)
+                    {
+                        osuInputManager.KeyBindingContainer.TriggerReleased(wasLeft ? OsuAction.RightButton : OsuAction.LeftButton);
+                        return;
+                    }
+
+                    osuInputManager.KeyBindingContainer.TriggerPressed(wasLeft ? OsuAction.LeftButton : OsuAction.RightButton);
+                    wasLeft = !wasLeft;
+                    return;
```

Comment:
I kinda don't like that this is basically the clone of the logic below but interfacing with the `KeyBindingContainer` rather than `ReplayState`... Maybe this could be addressed by extracting out a `Func<bool>` conditional on `legacyReplay`, but not sure. Maybe this means this direction isn't the best.

Reply:
It's not great, I agree. Maybe for the sake of consistent replay output we could make `KeyBindingContainer` responsible regardless of if the replay is legacy or not? Essentially, the key presses on a relax replay would not be relevant, this would also benefit the case where input generation changes.

Reply:
> Maybe for the sake of consistent replay output we could make `KeyBindingContainer` responsible regardless of if the replay is legacy or not?

I think doing this would cause inputs not to be written to the replay for lazer scores, and we do want that (even if for backwards compatibility at least, if we happen to substantially change how relax works somehow).

The func is probably the best that can be done for now I guess...

Reply:
Right... Yeah that makes sense, I'll just adjust the code to extract out a `Func<bool>`